Source/DafnyLanguageServer/Language/IVerificationProgressReporter.cs

```diff
@@ -0,0 +1,14 @@
+using System.Threading;
+
+namespace Microsoft.Dafny.LanguageServer.Language {
+  /// <summary>
+  /// A callback interface to report verification progress
+  /// </summary>
+  public interface IVerificationProgressReporter {
```

Comment:
This smells (in a good way :) ) like something we could make more structured and hence more useful for other purposes. As in, we could have multiple event payloads like `VerificationStarted(name)` and `VerificationComplete(name, result)`. Converting those into string messages would be something one possible reporter would do, but we could also use this as a better communication channel than parsing the Boogle `/xml` output for the `/verificationLogger` feature.

Reply:
Certainly.  That's what I was referring to in he commit message.  I was hoping this would be part of either your work on the verificationLogger feature or Remy's work on individual verification tasks.  If you want we can rescope this feature, but then I fear that the we'll step on each other's toes (?)